spread-shellcheck

```diff
@@ -160,16 +161,20 @@ def checkfile(path):
 
     if path.endswith('spread.yaml') and 'suites' in data:
         # check suites
-        for suite in data['suites'].keys():
-            for section in SECTIONS:
-                if section not in data['suites'][suite]:
-                    continue
+        with ThreadPoolExecutor() as executor:
+            futures = []
+            for suite in data['suites'].keys():
+                for section in SECTIONS:
+                    if section not in data['suites'][suite]:
+                        continue
+                        logging.debug("%s (suite %s): checking section %s", path, suite, section)
+                        futures.append(executor.submit(checksection, data['suites'][suite][section], env))
+            for future in futures:
                 try:
-                    logging.debug("%s (suite %s): checking section %s", path, suite, section)
-                    checksection(data['suites'][suite][section], env)
+                    future.result()
                 except ShellcheckRunError as serr:
                     errors.addfailure('suites/' + suite + '/' + section,
```

Comment:
This looks like it is going to attribute every failure to the last suite and section value from the prior for loops, rather than the values associated with the future.  Perhaps you should store `(suite, section, future)` in the futures list?

Reply:
Indeed, silly me!

Reply:
Fixed in https://github.com/snapcore/snapd/pull/9491